hoodie-client/src/main/java/com/uber/hoodie/io/compact/strategy/PartitionAwareCompactionStrategy.java

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Copyright (c) 2016 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+package com.uber.hoodie.io.compact.strategy;
+
+import com.uber.hoodie.config.HoodieWriteConfig;
+import com.uber.hoodie.exception.HoodieException;
+import com.uber.hoodie.io.compact.CompactionOperation;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+/**
+ * This strategy orders compactions in reverse order of creation of Hive Partitions.
+ * It helps to compact data in latest partitions first and then older capped at the Total_IO allowed.
+ *
+ */
+public class PartitionAwareCompactionStrategy extends BoundedIOCompactionStrategy {
+    private Comparator<CompactionOperation> comparator;
+
+    public PartitionAwareCompactionStrategy() {
+        super();
+        // For now, use SimpleDateFormat as default partition format
+        String datePartitionFormat = "yyyy/MM/dd";
```

Comment:
yeah this is a problem. 

We need a way to generalize this have some way of ordering partitions as a callback.

Reply:
That's why I exposed a constructor with a customComparator but using the date format as default.

Reply:
Ah got it - If there are other constructors always chain them. Do not call super() from more than one overloaded constructor.

Reply:
typically this is done by passing the comparator created in the constructor with no args to the one with an arg below (reverse of how its chained now) Also, can you define the default comparator as a static member with an explanation of what it does.